### PR TITLE
New approach to separating editor and content CSS

### PR DESCRIPTION
**Pull request by pszczesniak**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Fix (build-tools): Should preserved all needed selectors and at-rules in CSS files after split into editor and content stylesheets. Closes https://github.com/ckeditor/ckeditor5/issues/16703.

---

### Additional information

The main problem was when we were splitting main css file into `editor` and `content`. Mainly it was done using `.ck-content` CSS class as a main target what should be put into `content` file and all the rest should land into `editor`. But this condition didn't include neither `@media` queries or  animations with `@keyframes` that class contains.

First approach was to write it by us (PR: https://github.com/ckeditor/ckeditor5-dev/pull/972) but it was a lot of code and it wans't bullet proof for future. So decision was made to try something else: `PurgeCss` lib.

With using `PurgeCss` all at-rules are splitted correctly.

Also some CSS classes were omitted when there were more than one selector in definition. It's also fixed in this PR.

### CODE REVIEW

File: packages/ckeditor5-dev-build-tools/src/plugins/splitCss.ts [hunk 1]
```diff
@@ -30,10 +30,35 @@ export interface RollupSplitCssOptions {
...
+	content: [],
+	safelist: { deep: [ /^ck-content/ ] },
+	blocklist: []
+};
+
+const purgeCssOptionsForEditor = {
+	...commonPurgeCssOptions,
+	// Pseudo class`:where` is preserved only if the appropriate html structure matches the CSS selector.
+	// It's a temporary solution to avoid removing selectors for Show blocks styles where `:where` occurs.
+	// See: https://github.com/FullHuman/purgecss/issues/978
```
Comment (filipsobol) [inline packages/ckeditor5-dev-build-tools/src/plugins/splitCss.ts:50]: I'd be nice to know what **exact** issue this HTML solves.

File: packages/ckeditor5-dev-build-tools/src/plugins/splitCss.ts [hunk 2]
```diff
@@ -30,10 +30,35 @@ export interface RollupSplitCssOptions {
...
+	content: [ {
+		raw: '<html><body><div class="ck ck-editor__editable ck-editor__editable_inline ck-show-blocks">' +
```
Comment (filipsobol) [inline packages/ckeditor5-dev-build-tools/src/plugins/splitCss.ts:52]: Would PurgeCSS handle this HTML if we used [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) and format this HTML nicely?

File: packages/ckeditor5-dev-build-tools/src/plugins/splitCss.ts [hunk 3]
```diff
@@ -297,16 +170,21 @@ async function unifyFileContentOutput( content: string = '', minimize: boolean )
...
  */
-function wrapDefinitionsIntoSelector( selector: string, definitions: string ): string {
-	// When definition contains `data:image` it should be in one line following the specification.
-	// Currently used tool responsible for parsing definitions tries to split each definition into new line.
-	// When `data:image` contains `SVG` with style attribute which contains CSS definitions it splits it into new lines,
-	// which breaks the CSS.
-	if ( definitions.includes( 'data:image' ) ) {
-		definitions = removeNewline( definitions );
-	}
+async function normalizeStylesheet( content: string ): Promise<string> {
```
Comment (filipsobol) [inline packages/ckeditor5-dev-build-tools/src/plugins/splitCss.ts:175]: The name and description of this function don't tell me much about what it does.

File: packages/ckeditor5-dev-build-tools/src/plugins/splitCss.ts (hunk 1 above)
Comment (pszczesniak) [reply packages/ckeditor5-dev-build-tools/src/plugins/splitCss.ts]: Addressed in 96d3bb7.

File: packages/ckeditor5-dev-build-tools/src/plugins/splitCss.ts (hunk 2 above)
Comment (pszczesniak) [reply packages/ckeditor5-dev-build-tools/src/plugins/splitCss.ts]: Yes, it works with template literals. 96d3bb7

File: packages/ckeditor5-dev-build-tools/src/plugins/splitCss.ts (hunk 3 above)
Comment (pszczesniak) [reply packages/ckeditor5-dev-build-tools/src/plugins/splitCss.ts:184]: I've change the description of the function to show exactly whit it does.

**Review by filipsobol — APPROVED** [review]

**Pull request merged**